src/converse-httpFileUpload.js

```diff
@@ -0,0 +1,269 @@
+                                requestSlotUrl = jid;
+                                ready = true;
+                                self.sendFile();
+                            }
+                        }
+                    });
+                },
+
+                /**
+                 * Saves the file the user has picked.
+                 * 
+                 * @param {*} file the name of the file the user has picked.
+                 * @param {*} chatBox the chatbox from which the user initiated the file-upload
+                 */
+                setFile (file1, chatBox1){
+                    file = file1;
+                    chatBox = chatBox1;
+                    this.sendFile();
+                },
+                
+                /**
+                * Upload file.
+                * Waits till the Upload-Service is discovered and till the user has picked a file.
+                *
+                */
+                sendFile () {
+                    var self = this;
+                    if(file === undefined){
+                        console.log("waiting to choose a file");
+                        return;
+                    }
+                    else if(requestSlotUrl === undefined){
+                        console.log("waiting for service discovery");
+                        return;
+                    }
+
+                    console.log('Send file via http upload');
+                    chatBox.showHelpMessages([__('The file upload starts now')],'info');
+                    this.requestSlot(file, function(data) {
+                        if (!data) {
+                            // general error
+                            console.log('Unknown error while requesting upload slot.');
+                            alert(__('File upload failed. Please check the log.'));
+                        } else if (data.error) {
+                            // specific error
+                            console.log('The XMPP-Server return an error of the type: ' + data.error.type);
+                            alert(__('File upload failed. Please check the log.'));
+                        } else if (data.get && data.put) {
+                            console.log('slot received, start upload to ' + data.put);
+                            self.uploadFile(data.put, file, function() {
+                                console.log(data.put);
+                                
+                                chatBox.onMessageSubmitted(data.put, null, file);
+                                file = undefined;
+                            });
+                        }
+                    });
+                },
+
+                /**
+                * Request upload slot from xmpp-server
+                *
+                * @param  {File} file the file the user picked
+                * @param  {Function} cb Callback after finished request
+                */
```

Comment:
These docstrings don't conform to how docstrings are done in this project. In general the styleguide in this file doesn't conform to the [Converse.js Code Style Guide](https://conversejs.org/docs/html/style_guide.html).